### PR TITLE
CDPT-1680 Refactor Nselect

### PR DESCRIPTION
**Pull request by wilson1000**
There is nothing extreme in this PR. Just an update to the JS file that includes:

- Performance: caching more than 50 redeclared jQuery objects
- Readability and syntax improvements

Sadly, the update means the file is completely changed, this doesn't help with review. Sorry about that!

### CODE REVIEW

File: public/app/themes/justice/src/js/legacy/nselect.js [hunk 1]
```diff
@@ -1,149 +1,199 @@
...
+                        _nselect_ul.children('li.hover').removeClass('hover')
+                        jQuery(_nselect_ul.children('li')[ii]).addClass('hover')
+                        stop = true
+                        break
+                    }
+                }
+
+                if (!stop) {
+                    let iii = 0
+                    for (iii; iii <= current; iii++) {
```
Comment (EarthlingDavey) [inline public/app/themes/justice/src/js/legacy/nselect.js:145]: Wow, this is nice code.

No action required.

**Review by EarthlingDavey — APPROVED** [review]
Hey, looks good. Approved with a couple of comments.

It was quite hard to review the changes because of the change in spaces and line endings. If you reformat to match the original - I can re-review. Then you can re-change to 4 spaces w/o semi-colons. Your shout.

File: public/app/themes/justice/src/js/legacy/nselect.js [hunk 2]
```diff
@@ -1,149 +1,199 @@
...
+    let offset = 0
+
+    if (typeof (html) == 'undefined' || html.length === 0) {
+        html = jQuery('body')
+    } else {
+        offset = jQuery('[id^="nselect"]').length
+    }
+
+    jQuery(html).find('select').each(function (i) {
+        let _select = jQuery(this)
```
[the remark this answers is not in the document]
Comment (wilson1000) [reply public/app/themes/justice/src/js/legacy/nselect.js:18]: I use it to denote either, _private or _local - it's just a way I code.

File: public/app/themes/justice/src/js/legacy/nselect.js [hunk 3]
```diff
@@ -1,149 +1,199 @@
...
+        if (typeof _select.attr('class') != 'undefined' && _select.attr('class').indexOf('no-nselect') > -1) {
+            return
+        }
```
[the remark this answers is not in the document]
Comment (wilson1000) [reply public/app/themes/justice/src/js/legacy/nselect.js:21]: Thanks for pointing that out mate 👍

**Pull request merged**